Quadriga/src/test/java/edu/asu/spring/quadriga/service/impl/projectblog/ProjectBlogEntryManagerTest.java

```diff
@@ -0,0 +1,176 @@
+    private static final String DESCRIPTION_2 = "TITLE_2";
+
+    @Mock
+    private IProjectBlogEntryDAO projectBlogEntryDAO = Mockito.mock(IProjectBlogEntryDAO.class);
+
+    @Mock
+    private ProjectBlogEntryDTOMapper projectBlogEntryDTOMapper = Mockito.mock(ProjectBlogEntryDTOMapper.class);
+
+    @InjectMocks
+    private ProjectBlogEntryManager projectBlogEntryManagerUnderTest;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    /**
+     * tests
+     * {@link IProjectBlogEntryManager#getProjectBlogEntryList(String, Integer)}
+     * method responsible for retrieving list of project blog entries for a
+     * project
+     * 
+     * @throws QuadrigaStorageException
+     */
+    @Test
+    public void Test() throws QuadrigaStorageException {
+
+        // Creating dummy list of blog entry DTOs to return
+        List<ProjectBlogEntryDTO> projectBlogEntryDTOList = new ArrayList<>();
+        ProjectBlogEntryDTO blgeDTO1 = createProjectBlogEntryDTO(BLGE_ID_1, TITLE_1, DESCRIPTION_1);
+        projectBlogEntryDTOList.add(blgeDTO1);
+        ProjectBlogEntryDTO blgeDTO2 = createProjectBlogEntryDTO(BLGE_ID_2, TITLE_2, DESCRIPTION_2);
+        projectBlogEntryDTOList.add(blgeDTO2);
+
+        // Return the dummy list of blog entries created above when method to
+        // get project blog entries is called
+        Mockito.when(projectBlogEntryDAO.getProjectBlogEntryDTOListByProjectId(DUMMY_PROJECT_ID, LIMIT))
+                .thenReturn(projectBlogEntryDTOList);
+
+        // Creating dummy project blog entry
+        IProjectBlogEntry blge1 = new ProjectBlogEntry();
+        blge1.setProjectBlogEntryId(BLGE_ID_1);
+        blge1.setTitle(TITLE_1);
+        blge1.setDescription(DESCRIPTION_1);
+
+        Mockito.when(projectBlogEntryDTOMapper.getProjectBlogEntry(blgeDTO1)).thenReturn(blge1);
+
+        // Creating dummy project blog entry
+        IProjectBlogEntry blge2 = new ProjectBlogEntry();
+        blge2.setProjectBlogEntryId(BLGE_ID_2);
+        blge2.setTitle(TITLE_2);
+        blge2.setDescription(DESCRIPTION_2);
+
+        Mockito.when(projectBlogEntryDTOMapper.getProjectBlogEntry(blgeDTO2)).thenReturn(blge2);
+
+        // Calling the method to test
+        List<IProjectBlogEntry> projectBlogEntryList = projectBlogEntryManagerUnderTest
+                .getProjectBlogEntryList(DUMMY_PROJECT_ID, LIMIT);
+
+        // Asserting the details
+        assertProjectBlogEntryList(projectBlogEntryList);
+    }
+
+    private void assertProjectBlogEntryList(List<IProjectBlogEntry> projectBlogEntryList) {
+
+        // Assert that exactly 2 entries are fetched
+        assertEquals(projectBlogEntryList.size(), 2);
+
+        // Assert the details of first blog fetched only when atleast one entry
+        // is fetched
+        // If there are fewer entries than expected, first assert will fail and
+        // NullPointerException is avoided
+        if (projectBlogEntryList.size() >= 1) {
```

Comment:
why do you do these if statements here? you already made sure there are two elements in the list.